libs/vectorise/include/vectorise/fixed_point/fixed_point.hpp

```diff
@@ -932,6 +950,104 @@ constexpr FixedPoint<16, 16>::FixedPoint(FixedPoint<32, 32> const &o)
   }
 }
 
+template <uint16_t I, uint16_t F>
+FixedPoint<I, F>::FixedPoint(std::string const &s)
+  : data_{0}
+{
+  auto index  = s.find("fp");
+  auto s_copy = std::string(s, 0, index);
+  std::transform(s_copy.begin(), s_copy.end(), s_copy.begin(),
+                 [](char c) { return std::tolower(c); });
+
+  Type         integer_part{0};
+  UnsignedType fractional_part{0};
+  Type         exponent_part{1};
+
+  // first find exponent part if it exists
+  auto exponent_pos = s_copy.find('e');
+  if (exponent_pos != std::string::npos)
+  {
+    auto exponent_match = std::string(s_copy, exponent_pos + 1, s_copy.length());
+    exponent_part       = static_cast<Type>(std::strtoll(exponent_match.c_str(), nullptr, 10));
+    s_copy.erase(exponent_pos, s_copy.length() - exponent_pos);
+  }
+
+  std::string integer_match;
+  std::string fractional_match;
+  auto        decimal_pos = s_copy.find('.');
+  if (decimal_pos != std::string::npos)
+  {
+    integer_match    = std::string(s_copy, 0, decimal_pos);
+    fractional_match = std::string(s_copy, decimal_pos + 1, s_copy.length());
+    // Parse the fractional part and convert to FixedPoint.
+    // Ignore more than the supported decimal digits
+    if (fractional_match.length() > DECIMAL_DIGITS)
+    {
+      fractional_match.erase(DECIMAL_DIGITS, fractional_match.length() - DECIMAL_DIGITS);
+    }
+    // read into the integer type and divide by largest possible power of 10 to get
+    // the needed fractional part in binary
+    fractional_part =
+        static_cast<UnsignedType>(std::strtoul(fractional_match.c_str(), nullptr, 10));
+    UnsignedType power10{1};
+    while (power10 < fractional_part)
+    {
+      power10 *= 10;
+    }
+    fractional_part = (fractional_part * ONE_MASK) / power10;
+  }
+  else
+  {
+    integer_match = s_copy;
+  }
+
+  if (integer_match.length() > DECIMAL_DIGITS)
```

Comment:
what if the integer part is 00000000000001 ?

Reply:
hm, I thought the parser in the vm already catches those, will check if that's the case. I can add an extra zeros trimming if not.

Reply:
The tokeniser won't let through any superfluous trailing zeros. An integer part of a number is either exactly "0" or starts with a non zero digit

Reply:
added removal of leading and trailing zeros so this should be covered now.